cumulusci/salesforce_api/utils.py

```diff
@@ -12,12 +13,27 @@
 def get_simple_salesforce_connection(
     project_config, org_config, api_version=None, base_url: str = None
 ):
+
     # Retry on long-running metadeploy jobs
     retries = Retry(total=5, status_forcelist=(502, 503, 504), backoff_factor=0.3)
     adapter = HTTPAdapter(max_retries=retries)
+    instance_url = org_config.instance_url
+
+    # Attept to get the host and port from the URL, but ignore any errors retrieving it.
+    try:
+        instance_url = urlparse(org_config.instance_url).hostname
+    except:
+        pass
+
+    try:
+        port = urlparse(org_config.instance_url).port
+        if port:
+            instance_url = instance_url.replace(".com", ".com:" + str(port))
+    except:
+        pass
```

Comment:
Are there particular errors you're expecting? I think it should generally work to use urlparse to get the hostname and port as long as the input is actually a URL, and if it's not, that's probably an error we would not want to suppress.

Reply:
I had added the try/except blocks only because it was failing in the tests. It looks like the mock generates an invalid url that gets kicked back by the call to urlparse().

Reply:
We should probably handle that by setting up more realistic instance_urls in the tests, including when they are coming from a mock.